sandy/decay.py

```diff
@@ -148,6 +156,39 @@ def get_decayconstant(self, with_uncertainty=True):
         else:
             return df.LAMBDA
 
+    def custom_perturbation_decayconstant(self, pert, zam="all"):
+        """
+        Apply a custom perturbation to the decay constant
+        Parameters
+        ----------
+        zam : `int` or `list` or `all`, default is `all`
+            ZAM number of the material to which perturbation is to be
+            applied.
+        pert : 'float'
+            Perturation coeffcient as ratio value.
+        Returns
+        -------
+        `sandy.decay.DecayData`
+            DacayData instance with given value ZAM and decay constant perturbed.
+        Examples
+        --------
+        >>> endf6 = sandy.get_endf6_file("jeff_33",'decay',[922350, 922330, 942390, 942320, 922350])
+        >>> rdd = sandy.DecayData.from_endf6(endf6)
+        >>> tape =sandy.DecayData.custom_perturbation_decayconstant(rdd, 0.05, [922350, 922330])
+        >>> print(tape)
+        922350   3.27689e-17
+        922330   1.44821e-13
+        Name: LAMBDA, dtype: float64
+        """
+        tape = sandy.DecayData.get_decayconstant(self)
+        if zam != 'all':
+            for z in zam:
+                tape.LAMBDA[z] = tape.LAMBDA[z] * (1 + pert)
+            return self.__class__(tape.LAMBDA[zam])
+        else:
+            tape.LAMBDA = tape.LAMBDA * (1 + pert)
+            return self.__class__(tape.LAMBDA)

```

Comment:
The docstring says you should return a DecayData...I am not sure what you return, but it's wrong

Reply:
Do you prefer a pandas Series as outout? I printed the type of the output that I obtained and it said <class 'sandy.decay.DecayData'>, for this reason I wrote that in the docstring but I understand that I can do it differently

Reply:
No, as we mentioned a few times already, you should be able to to rewrite the DecayData instance with perturbed decay constants into an Endf6 instance and then to file.
At the moment you cannot do it because, despite you return a DecayData instance, it does not contain the right data...then all DecayData methods will fail, including `to_endf6`.

returning a pandas Series would just lead you even further away from the problem.

If you have problems in seeing the bigger picture, try and develop more steps on a notebook